tests/api/test_collect_runs_logs.py

```diff
@@ -182,6 +184,63 @@ async def test_start_log_for_run_failure(
         assert run_uid is None
         assert log_collector._call.call_count == 1
 
+    @pytest.mark.asyncio
+    async def test_log_collector_sidecar(
+        self, db: sqlalchemy.orm.session.Session, client: fastapi.testclient.TestClient
+    ):
+        # TODO: start log collector sidecar in test
+
+        # set config log collector sidecar values to enable it
+        prev_log_collector_mode = mlrun.config.config.log_collector.mode
+        prev_log_collector_address = mlrun.config.config.log_collector.address
+        mlrun.config.config.log_collector.mode = "sidecar"
+        mlrun.config.config.log_collector.address = "localhost:8282"
+
+        uid = "cde099c6724742859b8b2115eb767429"
+        _, _, uid, _, run = _create_new_run(
+            db,
+            "default",
+            uid=uid,
+            name="handler",
+            kind="job",
+            store=False,
+        )
+
+        # create an actual pod for the run
+        pod_name = "some-pod"
+        namespace = mlrun.api.utils.singletons.k8s.get_k8s().resolve_namespace()
+        if not namespace:
+            namespace = "mlrun"
+        _create_dummy_pod_for_run(pod_name, namespace, uid)
+
+        try:
+            run_uid = await mlrun.api.main._start_log_for_run(
+                run, self.start_log_limit, raise_on_error=False
+            )
+            assert run_uid == uid
+
+            # wait for the log collector to collect some logs
+            await asyncio.sleep(5)
+
+            logs = await mlrun.api.utils.clients.log_collector.LogCollectorClient().get_logs(
+                run_uid=uid, project="default", offset=0, size=-1
+            )
+            assert len(logs) < 95000
+
+            logs = await mlrun.api.utils.clients.log_collector.LogCollectorClient().get_logs(
+                run_uid=uid, project="default", offset=95000, size=-1
+            )
+
+            # offset is bigger than the log size, so we expect no logs
+            assert len(logs) == 0
+        finally:
+            mlrun.config.config.log_collector.mode = prev_log_collector_mode
+            mlrun.config.config.log_collector.address = prev_log_collector_address
+
+            # delete pod created for this test
+            k8s_helper = mlrun.api.utils.singletons.k8s.get_k8s_helper()
+            k8s_helper.delete_pod(pod_name, namespace)
```

Comment:
`tests/api` is part of our unit tests suites which only runs in docker environment. The `k8s` stuff needs to be mocked.